clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1481,12 +1494,12 @@ protected void checkAndHandleUpstreamOffsetRewind(
               int actualSchemaId = ByteUtils.readInt(actualValue, 0);
               Put put = (Put) envelope.payloadUnion;
               if (actualSchemaId == put.schemaId) {
-                // continue if schema Id is the same
-                if (ByteUtils.equals(
-                    put.putValue.array(),
-                    put.putValue.position(),
-                    actualValue,
-                    ValueRecord.SCHEMA_HEADER_LENGTH)) {
+
+                if (put.putValue.equals(
+                    ByteBuffer.wrap(
```

Comment:
Seems like the issue with this `ByteUtils::equals` function is that it doesn't take into account the end (e.g. limit of the BB)? Can we update this function to add end offset parameters instead of allocating a new BB?

Also, this is the only usage of this function, so if instead of updating it we stop using it, then it should be deleted...

Reply:
I guess we can delete that function since it is incorrect and duplicate from the ByteBuffer.equals function.
The reasons I thought it is fine to use `ByteBuffer#equals`:
1. `ByteBuffer` is a thin wrapper of byte[].
2. `ByteBuffer#equals` has a correct impl and it is not worth maintaining a duplicate logic in our code base.
3. This function is not in hot path.

Reply:
Is it true that it's not on the hot path? I seem to recall that when these split brain logs come up, they can be quite voluminous, which seems to indicate it can be somewhat hot, at least for a little bit. I could be mis-remembering though.

Reply:
Split brain issue is still rare comparing to the regular execution and as you mentioned, it will only be executed when this issue happens for that period.